plugins/aws-appmesh/plugin/commands.go

```diff
@@ -226,6 +226,11 @@ func (plugin *Plugin) setupIngressRules(
 		return err
 	}
 
+	// When no app port specified, appPorts field is a string array of length 1 with an empty string element.
+	if len(config.AppPorts) > 0 && config.AppPorts[0] == "" {
```

Comment:
Why do we not handle this case in netconfig.go? If the array is an array of length 1 with an empty element, just set it to nil in netconfig while validating the input so that you don't have to keep checking for that multiple times during execution.

Reply:
My original thought was not to change the value in validation process but yeah put them in one place is cleaner.